packages/lesswrong/components/walledGarden/PortalBarGcalEventItem.tsx

```diff
@@ -17,30 +17,34 @@ const styles = (theme) => ({
     width: 120,
     textAlign: "right",
     display: "inline-block"
-  },
-  tooltip: {
-    whiteSpace: "pre-wrap"
   }
 })
 
-const UrlClass = getUrlClass()
 
-const PortalBarGcalEventItem = ({classes, gcalEvent}) => {
+export const getAddToCalendarLink = (gcalEvent) => {
   const { LWTooltip } = Components
-
+  
+  const UrlClass = getUrlClass()
   const url = new UrlClass(gcalEvent.htmlLink)
   const eid = url.searchParams.get("eid")
   const addToCalendarLink = `https://calendar.google.com/event?action=TEMPLATE&tmeid=${eid}&tmsrc=${gcalEvent.organizer.email}`
```

Comment:
`eid` and `gcalEvent.organizer.email` need sanitization. I'm guessing they should be URL encoded with `encodeURIComponent`, but I'm not sure? They at least need to exclude `&` and `#`.

Reply:
Pretty sure `eid` is sanitized. Given that it comes right out of the browser API for `searchParams`. The organizer email does seem like it probably needs sanitization.

Reply:
Nope--`searchParams` decodes it as a URI parameter, which means it needs to be re-encoded before putting it into another URI.

Reply:
```
const u = new URL("http://www.example.com/?nastyParam="+encodeURIComponent("&differentParam=1"))
u.searchParams.get("nastyParam")
```
=> `&differentParam=1`

Reply:
Ah, yeah. You're right.